### PR TITLE
harfbuzz: Remove redundant cairo dependency

### PR DESCRIPTION
**Pull request by roberth**
###### Motivation for this change

This removes transitive (build) dependencies on various X11 or
other GUI-related libraries, which is desirable for minimizing
rebuilds and maximizing parallellism.

This also adds fontconfig to buildInputs, because this direct lib
dependency was previously found via cairo's propagatedBuildInputs.

To be sure, I've tested harfbuzz by building a gtk app and pasting Indic and Thai text samples from the harfbuzz test data into an input field.

Found in an attempt to reduce the build closure of `pythonPackages.sphinx` #126611 

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 126659` at b93297e9 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>186 packages marked as broken and skipped:</summary>
  <ul>
    <li>adoptopenjdk-hotspot-bin-13</li>
    <li>adoptopenjdk-hotspot-bin-14</li>
    <li>adoptopenjdk-jre-hotspot-bin-13</li>
    <li>adoptopenjdk-jre-hotspot-bin-14</li>
    <li>airtame</li>
    <li>ams-lv2</li>
    <li>artyFX</li>
    <li>asymptote</li>
    <li>autotrace</li>
    <li>betaflight-configurator</li>
    <li>bless</li>
    <li>cargo-rr</li>
    <li>cassandra_2_1</li>
    <li>cassandra_2_2</li>
    <li>colmapWithCuda</li>
    <li>cura_stable</li>
    <li>deeptools</li>
    <li>devede</li>
    <li>displaycal</li>
    <li>dvd-slideshow</li>
    <li>electron_3</li>
    <li>electron_4</li>
    <li>electron_5</li>
    <li>embree</li>
    <li>embree2</li>
    <li>evilvte</li>
    <li>exodus</li>
    <li>ffmpeg_2</li>
    <li>ffmpeg_2_8</li>
    <li>ffmpeg_3</li>
    <li>ffmpeg_3_4</li>
    <li>firestarter</li>
    <li>forge</li>
    <li>gb-backup</li>
    <li>git-dit</li>
    <li>globalplatform</li>
    <li>gmtk</li>
    <li>gnome_mplayer</li>
    <li>gnuradio3_7</li>
    <li>gnuradio3_7Packages.ais</li>
    <li>gnuradio3_7Packages.gnuradio</li>
    <li>gnuradio3_7Packages.gsm</li>
    <li>gnuradio3_7Packages.limesdr</li>
    <li>gnuradio3_7Packages.nacl</li>
    <li>gnuradio3_7Packages.osmosdr</li>
    <li>gnuradio3_7Packages.rds</li>
    <li>gopro</li>
    <li>gppcscconnectionplugin</li>
    <li>gpshell</li>
    <li>grass</li>
    <li>green-pdfviewer</li>
    <li>gtkd</li>
    <li>hplipWithPlugin_3_16_11</li>
    <li>hplipWithPlugin_3_18_5</li>
    <li>hplip_3_16_11</li>
    <li>hplip_3_18_5</li>
    <li>impressive</li>
    <li>intecture-cli</li>
    <li>intel-compute-runtime</li>
    <li>invoice2data</li>
    <li>isabelle</li>
    <li>jboss</li>
    <li>jfbpdf</li>
    <li>jfbview</li>
    <li>k2pdfopt</li>
    <li>k3d</li>
    <li>keyfinder</li>
    <li>libgroove</li>
    <li>libvmi</li>
    <li>linuxPackages_hardkernel_4_14.akvcam</li>
    <li>linuxPackages_hardkernel_4_14.dpdk</li>
    <li>linuxPackages_hardkernel_latest.akvcam</li>
    <li>linuxPackages_hardkernel_latest.dpdk</li>
    <li>linuxPackages_testing_bcachefs.akvcam</li>
    <li>linuxPackages_testing_bcachefs.dpdk</li>
    <li>loxodo</li>
    <li>luxcorerender</li>
    <li>magma</li>
    <li>mame</li>
    <li>meo</li>
    <li>metamorphose2</li>
    <li>mupdf_1_17</li>
    <li>mvapich</li>
    <li>natron</li>
    <li>nixui</li>
    <li>nvtop</li>
    <li>obs-ndi</li>
    <li>ocropus</li>
    <li>odpdown</li>
    <li>openems</li>
    <li>openrw</li>
    <li>opentsdb</li>
    <li>pcsxr</li>
    <li>pdfdiff</li>
    <li>pgadmin</li>
    <li>photoflow</li>
    <li>pig</li>
    <li>plover.stable</li>
    <li>poedit</li>
    <li>polymake</li>
    <li>processing</li>
    <li>pulseeffects-legacy</li>
    <li>pulseeffects-pw</li>
    <li>python38Packages.HTSeq</li>
    <li>python38Packages.caffe</li>
    <li>python38Packages.cnvkit</li>
    <li>python38Packages.dask-gateway</li>
    <li>python38Packages.dask-jobqueue</li>
    <li>python38Packages.dask-ml</li>
    <li>python38Packages.dask-mpi</li>
    <li>python38Packages.dask-xgboost</li>
    <li>python38Packages.datashader</li>
    <li>python38Packages.fipy</li>
    <li>python38Packages.foundationdb51</li>
    <li>python38Packages.foundationdb52</li>
    <li>python38Packages.foundationdb60</li>
    <li>python38Packages.foundationdb61</li>
    <li>python38Packages.foxdot</li>
    <li>python38Packages.ignite</li>
    <li>python38Packages.pycuda</li>
    <li>python38Packages.pygmt</li>
    <li>python38Packages.pynvml</li>
    <li>python38Packages.python-csxcad</li>
    <li>python38Packages.python-openems</li>
    <li>python38Packages.pytorchWithCuda</li>
    <li>python38Packages.spacy</li>
    <li>python38Packages.spacy-transformers</li>
    <li>python38Packages.streamz</li>
    <li>python38Packages.stumpy</li>
    <li>python38Packages.tensorflowWithCuda</li>
    <li>python38Packages.textacy</li>
    <li>python39Packages.HTSeq</li>
    <li>python39Packages.caffe</li>
    <li>python39Packages.cnvkit</li>
    <li>python39Packages.dask-gateway</li>
    <li>python39Packages.dask-jobqueue</li>
    <li>python39Packages.dask-mpi</li>
    <li>python39Packages.dask-xgboost</li>
    <li>python39Packages.fipy</li>
    <li>python39Packages.foundationdb51</li>
    <li>python39Packages.foundationdb52</li>
    <li>python39Packages.foundationdb60</li>
    <li>python39Packages.foundationdb61</li>
    <li>python39Packages.foxdot</li>
    <li>python39Packages.ignite</li>
    <li>python39Packages.pycuda</li>
    <li>python39Packages.pygmt</li>
    <li>python39Packages.pynvml</li>
    <li>python39Packages.python-csxcad</li>
    <li>python39Packages.python-openems</li>
    <li>python39Packages.pytorchWithCuda</li>
    <li>python39Packages.streamz</li>
    <li>python39Packages.tensorflowWithCuda</li>
    <li>qemu_xen</li>
    <li>qemu_xen-light</li>
    <li>qemu_xen_4_10</li>
    <li>qemu_xen_4_10-light</li>
    <li>qlandkartegt</li>
    <li>qubes-core-vchan-xen</li>
    <li>ring-daemon</li>
    <li>scribus</li>
    <li>shotgun</li>
    <li>sonic-pi</li>
    <li>spark</li>
    <li>split2flac</li>
    <li>spotifywm</li>
    <li>spring</li>
    <li>springLobby</li>
    <li>stepmania</li>
    <li>swift</li>
    <li>termplay</li>
    <li>tilix</li>
    <li>tlaps</li>
    <li>torchat</li>
    <li>truvari</li>
    <li>tvheadend</li>
    <li>twinkle</li>
    <li>unetbootin</li>
    <li>unicap</li>
    <li>vbam</li>
    <li>vcs</li>
    <li>winpdb</li>
    <li>xpdf</li>
    <li>yaxg</li>
    <li>yosys-bluespec</li>
    <li>zandronum</li>
  </ul>
</details>
<details>
  <summary>14253 packages skipped due to time constraints:</summary>
  <ul>
    <li>AusweisApp2</li>
    <li>DisnixWebService</li>
    <li>EBTKS</li>
    <li>Fabric (python38Packages.Fabric)</li>
    <li>MIDIVisualizer</li>
    <li>R</li>
    <li>SDL_Pango</li>
    <li>SDL_mixer</li>
    <li>abcl</li>
    <li>abcm2ps</li>
    <li>...</li>
  </ul>
</details>
<details>
  <summary>5 packages built successfully:</summary>
  <ul>
    <li>asciidoctor</li>
    <li>pango (gnome2.pango)</li>
    <li>harfbuzz</li>
    <li>lasem</li>
    <li>libass</li>
  </ul>
</details>

File: pkgs/development/libraries/harfbuzz/default.nix [hunk 1]
```diff
@@ -4,7 +4,7 @@
 , pkg-config
 , glib
 , freetype
-, cairo
+, fontconfig
```
Comment (khaledhosny) [inline pkgs/development/libraries/harfbuzz/default.nix:7]: HarfBuzz does not have a direct dependency on FontConfig (the configure script used to check for it optionally, but it was actually unused).

Comment (roberth) [reply pkgs/development/libraries/harfbuzz/default.nix:7]: Thank you for fixing the configure script _and_ pointing this out. :+1: 

I only had to update it to remove `fontconfig`.

**Review by SuperSandro2000 — APPROVED** [review]
diff LGTM, don't really know anything about the actual change

**Pull request merged**